### PR TITLE
CLOUDP-70013: add IAM users list

### PR DESCRIPTION
**Pull request by robcarlan-mlab**
## Proposed changes

_Jira ticket:_ CLOUDP-70013

Closes #[issue number]

## Checklist

- [] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have added any necessary documentation (if appropriate)
- [] I have run `make fmt` and formatted my code

## Further comments

OM:
```
./bin/mongocli iam projects users list --profile om
ID                         FIRST NAME   LAST NAME   USERNAME
5f4579d04bba012f25a03212   rob          carlan      rob.carlan@mongodb.com
```

Atlas:
```
./bin/mongocli iam projects users list
ID                         FIRST NAME   LAST NAME   USERNAME
5f46a38224b4da52a6f4d8bd   Rob2         LastName    rob.carlan+new@mongodb.com
5c80f75fcf09a246878f67a4   Rob          Carlan      rob.carlan@mongodb.com
```

### CODE REVIEW

File: internal/store/projects.go [hunk 1]
```diff
@@ -113,3 +117,17 @@ func (s *Store) DeleteProject(projectID string) error {
 		return fmt.Errorf("unsupported service: %s", s.service)
 	}
 }
+
+// Users lists all IAM users in a project
+func (s *Store) Users(projectID string, opts *atlas.ListOptions) (interface{}, error) {
```
Comment (gssbzn) [inline internal/store/projects.go:122]: [nit] since we're going to have organization users and any user commands this may be better named `ProjectUsers`

**Review by gssbzn — CHANGES_REQUESTED** [review]
One small nit, also could we add an e2e test for this?

File: e2e/atlas/users_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+		if err != nil {
+			t.Fatalf("unexpected error: %v, resp: %v", err, string(resp))
+		}
+
+		var users []mongodbatlas.AtlasUser
+		if err := json.Unmarshal(resp, &users); err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if len(users) > 0 {
```
Comment (gssbzn) [inline e2e/atlas/users_test.go:53]: I think there's going to be users there

**Review by gssbzn — APPROVED** [review]
LGTM if LGTEVG (I think your e2e test may fail) thanks for doing all this

File: e2e/atlas/users_test.go (hunk 2 above)
Comment (robcarlan-mlab) [reply e2e/atlas/users_test.go:53]: ah probably yeah, I was thinking of adding the invite test before this and rearranging in that PR

**Review by gssbzn — APPROVED** [review]

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**